packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -820,23 +823,57 @@ export class DeltaManager
             this.deltaStorage = await docService.connectToDeltaStorage();
         }
 
-        const stream = this.deltaStorage.fetchMessages(
-            from, // inclusive
-            to, // exclusive
-            this.closeAbortController.signal,
-            cacheOnly);
-
-        // eslint-disable-next-line no-constant-condition
-        while (true) {
-            const result = await stream.read();
-            if (result.done) {
-                break;
+        let controller = this.closeAbortController;
+        let listenerToClear: ((op: ISequencedDocumentMessage) => void) | undefined;
+
+        if (to !== undefined) {
+            controller = new AbortController();
+
+            assert(this.closeAbortController.signal.onabort === null, "reentrancy");
```

Comment:
can you generate the shortcode in main for this, npm run policy-check:fix, and bring that with the port?